src/backend/api/handlers/decorators.py

```diff
@@ -15,26 +16,43 @@ def decorated_function(*args, **kwargs):
         auth_key = request.headers.get(
```

Comment:
cc @fangeugene This probably the only interesting bit if you wanna take a look

Reply:
`current_user()` doesn't make any DB calls if the session cookie is empty, right? Just want to make sure that we're not hitting the db excessively for normal API calls.

Reply:
Right. If we don't have a cookie *or* if the cookie is invalid, we don't hit the database. We only hit the database if we have a valid cookie.

We could modify this code path as well to only attempt to look for a user if we don't have an auth token. I kinda started in one place and ended in another on this PR - so it might be worth a second pass that does that optimization.

Reply:
I filed a tracking issue for this, because this is a good thought, and I don't want to lose sight of it. I've got an idea of how we might go about designing this as well. For now - I wanna try to work on some more pressing issues we have in the codebase. https://github.com/the-blue-alliance/the-blue-alliance/issues/3386

Reply:
I updated this code to fallback to fetching a user object only if an auth key is not used. That way it's clearer that we won't be doing DB calls if they're not absolutely necessary without needing to understand how `current_user` works under the hood. Good call on this 👍